### DISCUSSION TITLE
Proximity searching. Maximum distance between words/phrases. PG FTS '<#>' equivalent

### DISCUSSION THREAD

**Discussion by C-monC** [pgroonga/pgroonga#380] [Q&A] [answered]
Hi,

It appears as if it is not possible to do proximity queries in Pgroonga. Is that correct?
The Postgres full text function for proximity searching:

> tsquery <-> tsquery → tsquery
>              Constructs a phrase query, which matches if the two input queries match at successive lexemes.
>              to_tsquery('fat') <-> to_tsquery('rat') → 'fat' <-> 'rat'

I would like to search for where words are within 5 words of each other i.e. (<-> OR <2> OR <3> OR <4> OR <5>).

**Comment by kou** [comment, marked as answer]
It seems that `column &@~ '*NP30,0,5|5|5|5"A B C D E"'` is what you want.

See also: https://groonga.org/docs/reference/grn_expr/query_syntax.html#near-phrase-search-condition

**Reply by C-monC** [reply]
Thanks for the quick answer! This is exactly what I wanted.
